server/src/main/java/io/druid/segment/realtime/firehose/PredicateFirehose.java

```diff
@@ -31,6 +32,10 @@
  */
 public class PredicateFirehose implements Firehose
 {
+  private static final Logger log = new Logger(PredicateFirehose.class);
+  private static final int IGNORE_THRESHOLD = 5000;
+  private static long ignored = 0;
```

Comment:
also this should really be int and should be reset to 0 when crosses IGNORE_THRESHOLD, no reason to count the number of rows passing through here really.